pkg/apis/core/validation/validation.go

```diff
@@ -5156,9 +5156,6 @@ func ValidateNodeUpdate(node, oldNode *core.Node) field.ErrorList {
 		allErrs = append(allErrs, field.Forbidden(field.NewPath("spec", "providerID"), "node updates may not change providerID except from \"\" to valid"))
 	}
 
-	if node.Spec.ConfigSource != nil {
```

Comment:
keep the API validation as long as the field can be populated

Reply:
should we also log warnings when it's set or anything like that

Reply:
yeah, returning a warning that it's deprecated (and non-functional in 1.24+) sounds excellent

Reply:
(that's in nodeStrategy WarningsOnCreate/WarningsOnUpdate)

Reply:
updated the message in those. Reverted the change